pkg/deploy/job.go

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package deploy
+
+import (
+	"context"
+
+	orgv1 "github.com/eclipse/che-operator/pkg/apis/org/v1"
+	"github.com/eclipse/che-operator/pkg/util"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// SyncJobToCluster deploys new instance of given job
+func SyncJobToCluster(instance *orgv1.CheCluster, job *batchv1.Job, clusterAPI ClusterAPI) error {
+	if err := controllerutil.SetControllerReference(instance, job, clusterAPI.Scheme); err != nil {
+		return err
+	}
+
+	jobFound := &batchv1.Job{}
+	err := clusterAPI.Client.Get(context.TODO(), types.NamespacedName{Name: job.Name, Namespace: job.Namespace}, jobFound)
+	if err != nil && errors.IsNotFound(err) {
+		logrus.Infof("Creating a new object: %s, name: %s", job.Kind, job.Name)
+		err = clusterAPI.Client.Create(context.TODO(), job)
+		if err != nil {
+			logrus.Errorf("Failed to create %s %s: %s", job.Name, job.Kind, err)
+			return err
+		}
+		return nil
+	} else if err != nil {
+		logrus.Errorf("An error occurred: %s", err)
+		return err
+	}
+	return nil
+}
+
+// NewJob creates new job configuration by giben parameters.
+func NewJob(checluster *orgv1.CheCluster, name string, namespace string, image string, serviceAccountName string, env map[string]string, backoffLimit int32) *batchv1.Job {
```

Comment:
we can get namespace from checluster, not needed to pass it

Reply:
we don't configure ttlSecondsAfterFinished,
backoffLimit, why do we need to configure it? let's use some default value for instance 5.

Reply:
Just for case when we need to start a job in different than Che namespace.

Reply:
We indeed don't configure `ttlSecondsAfterFinished` so I may leave it with default value. As for `backoffLimit`, to me it may vary for different jobs, so I left it configurable. I would like to use some default values, but golang doesn't provide such ability.